app/views/booking_requests/confirmation_step.html.erb

```diff
@@ -33,9 +33,16 @@
     <strong><%= prisoner_step.prison_name %></strong>
   </p>
 
-  <p class="edit-link">
-    <%= link_to(t('.change_prisoner_details'), '/TODO') %>
-  </p>
+  <%= form_for(confirmation_step, url: booking_requests_path) do |f| %>
+    <%= render(partial: 'hidden_prisoner_step') %>
+    <%= render(partial: 'hidden_visitors_step') %>
+    <%= render(partial: 'hidden_slots_step') %>
+    <%= hidden_field_tag('review_step', 'prisoner_step') %>
+
+    <p class="edit-link">
+      <%= f.submit(t('.change_prisoner_details'), class: 'button') %>
+    </p>
```

Comment:
This is so much better, passing the state around this way rather than implicitly using cookies and so on! :boom: